pygmt/tests/test_grdcontour.py

```diff
@@ -22,103 +21,88 @@ def fixture_grid():
     return load_earth_relief(registration="gridline")
 
 
-@check_figures_equal()
+@pytest.mark.mpl_image_compare
 def test_grdcontour(grid):
     """
     Plot a contour image using an xarray grid with fixed contour interval.
     """
-    fig_ref, fig_test = Figure(), Figure()
-    kwargs = dict(interval="1000", projection="W0/6i")
-    fig_ref.grdcontour("@earth_relief_01d_g", **kwargs)
-    fig_test.grdcontour(grid, **kwargs)
-    return fig_ref, fig_test
+    fig = Figure()
+    kwargs = dict(interval="1000", projection="W0/15c", frame=True)
+    fig.grdcontour(grid, **kwargs)
+    return fig
 
 
-@check_figures_equal()
+@pytest.mark.mpl_image_compare
 def test_grdcontour_labels(grid):
     """
     Plot a contour image using a xarray grid with contour labels and alternate
     colors.
     """
-    fig_ref, fig_test = Figure(), Figure()
+    fig = Figure()
     kwargs = dict(
         interval="1000",
         annotation="5000",
-        projection="W0/6i",
+        projection="W0/15c",
         pen=["a1p,red", "c0.5p,black"],
-        label_placement="d3i",
+        label_placement="d6c",
+        frame=True,
     )
-    fig_ref.grdcontour("@earth_relief_01d_g", **kwargs)
-    fig_test.grdcontour(grid, **kwargs)
-    return fig_ref, fig_test
+
+    fig.grdcontour(grid, **kwargs)
+    return fig
 
 
-@check_figures_equal()
+@pytest.mark.mpl_image_compare
 def test_grdcontour_slice(grid):
     """
     Plot an contour image using an xarray grid that has been sliced.
     """
-
-    fig_ref, fig_test = Figure(), Figure()
+    fig = Figure()
 
     grid_ = grid.sel(lat=slice(-30, 30))
-    kwargs = dict(interval="1000", projection="M6i")
-    fig_ref.grdcontour(
-        grid="@earth_relief_01d_g", region=[-180, 180, -30, 30], **kwargs
-    )
-    fig_test.grdcontour(grid=grid_, **kwargs)
-    return fig_ref, fig_test
+    kwargs = dict(interval="1000", projection="M15c", frame=True)
+
+    fig.grdcontour(grid=grid_, **kwargs)
+    return fig
 
 
 @pytest.mark.mpl_image_compare
-def test_grdcontour_file():
+def test_grdcontour_file(grid):
```

Comment:
This test check if grdcontour works for a grid file input, so shouldn't be changed to `grid`.

Reply:
Thanks!